generate_serpent_chain.py

```diff
@@ -0,0 +1,180 @@
+    out = []
+    EXPECTED_ASSERT = 1
+    counted_assert = 0
+    with decpath.open("r") as stream:
+        while True:
+            try:
+                ev = Evaluation(stream)
+            except AssertionError:
+                if counted_assert < EXPECTED_ASSERT:
+                    counted_assert += 1
+                    continue
+                raise
+            except ValueError:
+                break
+            if ev.gnd_name == "Co55":
+                # Fix negative uncertainty
+                sec = ev.section[8, 457]
+                ev.section[8, 457] = "0".join([sec[:762], sec[764:]])
+            out.append(ev)
+    print(f"Found {len(out)} items in {decpath}")
+    print(f"  First: {out[0]}\n  Last: {out[-1]}")
+    return out
+
+
+def get_fpy_evals(fpypath: Path):
+    """Obtain neutron-induced fission product yields from file"""
+    # Can't use get_evaluations or else only one is returned
+    out = []
+    with fpypath.open("r") as stream:
+        while stream:
+            try:
+                ev = Evaluation(stream)
+            except ValueError:
+                break
+            out.append(ev)
+    print(f"Found {len(out)} items in {fpypath}")
+    print(f"  First: {out[0]}\n  Last: {out[-1]}")
+    return out
+
+
+class CustomFormatter(
+    argparse.ArgumentDefaultsHelpFormatter,
+    argparse.RawDescriptionHelpFormatter,
+):
+    pass
+
+
+parser = argparse.ArgumentParser(
+    description="""Generate an equivalent depletion chain from Serpent data""",
+    formatter_class=CustomFormatter,
+)
+
+parser.add_argument(
+    "nfy", type=Path, help="Path to neutron-induced fission yield file",
+)
+
+parser.add_argument(
+    "dec", type=Path, help="Path to radioactive decay file",
+)
+
+parser.add_argument(
+    "--neutron-dir",
+    type=Path,
+    help=(
+        "Path to ENDF-formatted incident neutron data. Defaults to pull "
+        "data from OPENMC_ENDF_DATA if not provided"
+    ),
+)
+
+
+parser.add_argument(
+    "--branches-from",
+    type=Path,
+    help="Pull branching ratios from existing XML depletion chain",
+)
+
+parser.add_argument(
+    "-o",
+    "--output",
+    type=Path,
+    help="Write new depletion chain to this file",
+    default="serpent_chain.xml",
+)
+
+args = parser.parse_args()
+
+if not args.nfy.is_file():
+    raise FileNotFoundError(args.nfy)
+
+if not args.dec.is_file():
+    raise FileNotFoundError(args.dec)
+
+if args.neutron_dir is None:
+    endfd = os.environ.get("OPENMC_ENDF_DATA")
+    if endfd is None:
+        raise EnvironmentError(
+            "Need to pass neutron directory or set OPENMC_ENDF_DATA "
+            "environment variable to find neutron reaction data"
+        )
+    neutron_dir = Path(endfd) / "neutrons"
+else:
+    neutron_dir = args.neutron_dir
+
+if not neutron_dir.is_dir():
+    raise NotADirectoryError(neutron_dir)
+
+if args.output.exists() and not args.output.is_file():
+    raise IOError(f"{args.output} exists but is not a file")
```

Comment:
IOError no longer exists (if you go to a Python prompt and type IOError, it will give you OSError)
```suggestion
    raise OSError(f"{args.output} exists but is not a file")
```